python/py-treq/Portfile

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-treq
+version             18.6.0
+revision            0
+
+maintainers         nomaintainer
+description         A requests-like API built on top of twisted.web's Agent
+long_description    ${description}
+
+platforms           darwin
+supported_archs     noarch
+
+homepage            https://github.com/twisted/treq
+master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
+distname            ${python.rootname}-${version}
+
+license             MIT
+
+checksums           sha256  91e09ff6b524cc90aa5e934b909c8d0d1a9d36ebd618b6c38e37b17013e69f48 \
+                    rmd160  0960e004dc9769df3228915bba439acc41162294 \
+                    size    59156
+
+python.versions     37
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
+    depends_lib-append \
+                    port:py${python.version}-attrs \
+                    port:py${python.version}-incremental \
+                    port:py${python.version}-requests \
+                    port:py${python.version}-six \
+                    port:py${python.version}-Twisted
+    
+    depends_test-append \
```

Comment:
Since the test dependencies are listed: did you also manage to run the unit tests (`test.run yes`, but maybe more is needed)?

Reply:
I tried adding this:
```
test.run            yes
test.env            PYTHONPATH=.
test.cmd            ${prefix}/bin/trial
test.target         treq
```
but it is giving an error:
`Error: Failed to test py-treq: command execution failed`

tail of log:
```
:debug:test test phase started at Mon Jun 10 19:24:26 IST 2019
:notice:test --->  Testing py-treq
:debug:test Executing org.macports.test (py-treq)
:debug:test Creating test directory: /opt/local/var/macports/build/_Users_rajdeep_portfiles_python_py-treq/py-treq/work/treq-18.6.0
:debug:test Environment: 
:debug:test CC_PRINT_OPTIONS='YES'
:debug:test CC_PRINT_OPTIONS_FILE='/opt/local/var/macports/build/_Users_rajdeep_portfiles_python_py-treq/py-treq/work/.CC_PRINT_OPTIONS'
:debug:test CPATH='/opt/local/include'
:debug:test LIBRARY_PATH='/opt/local/lib'
:debug:test MACOSX_DEPLOYMENT_TARGET='10.14'
:debug:test PYTHONPATH='.'
:info:test Executing:  cd "/opt/local/var/macports/build/_Users_rajdeep_portfiles_python_py-treq/py-treq/work/treq-18.6.0" && /opt/local/bin/trial treq 
:debug:test system:  cd "/opt/local/var/macports/build/_Users_rajdeep_portfiles_python_py-treq/py-treq/work/treq-18.6.0" && /opt/local/bin/trial treq 
:info:test sh: /opt/local/bin/trial: No such file or directory
:info:test Command failed:  cd "/opt/local/var/macports/build/_Users_rajdeep_portfiles_python_py-treq/py-treq/work/treq-18.6.0" && /opt/local/bin/trial treq 
:info:test Exit code: 127
:error:test Failed to test py-treq: command execution failed
:debug:test Error code: CHILDSTATUS 42070 127
:debug:test Backtrace: command execution failed
:debug:test     while executing
:debug:test "system {*}$notty {*}$nice $fullcmdstring"
:debug:test     invoked from within
:debug:test "command_exec test"
:debug:test     (procedure "porttest::test_main" line 4)
:debug:test     invoked from within
:debug:test "$procedure $targetname"
:error:test See /opt/local/var/macports/logs/_Users_rajdeep_portfiles_python_py-treq/py-treq/main.log for details.
```

Reply:
You have to use the version of ```trial``` that is installed for the same Python version - the easiest way to do this is as follows:
```
test.run        yes
test.cmd        trial-${python.branch}
test.target     treq
test.env        PYTHONPATH=${worksrcpath}/build/lib
```

it will result in some failures / reports of missing packages though....

This is likely related to the definition of the library dependency in setup.py as: ```Twisted[tls]```; I don't think that ```upt``` parses this nor that it's feasible to do so (ping @Korusuke @Steap). 
If you check what's needed for the [tls] part you'll see that you also need:
```
    tls=[
        'pyopenssl >= 16.0.0',
        # service_identity 18.1.0 added support for validating IP addresses in
        # certificate subjectAltNames
        'service_identity >= 18.1.0',
        # idna 2.3 introduced some changes that break a few things.  Avoid it.
        # The problems were fixed in 2.4.
        'idna >= 0.6, != 2.3',
],
```

Reply:
I changed the test commands to the above, but after running portindex, it says:
```
Failed to parse file python/py-treq/Portfile: can't read "python.branch": can't read "python.version": no such variable
```

Reply:
I tried out the ```test``` part locally and it does work... I haven't added the other dependencies , but I suspect you made a typo somewhere ;) You didn't update the PR with your changes so I cannot really tell what the issue could be.

Reply:
also, please sort the dependencies with your newest additions alphabetically

Reply:
Done